src/autopas/containers/linkedCells/traversals/LCC08CellHandlerUtility.cpp

```diff
@@ -0,0 +1,189 @@
+
+  // Small constants used multiple times in the code below
+  // Note, Legacy: With ov1 --> No support for assymetric overlaps!
+  const int ov1 = overlap[0] + 1;
+  const double interactionLengthSquare{interactionLength * interactionLength};
+
+  // Due to 2D output, we need to first resize the outer vector to the x dimension
+  if constexpr (Mode == C08OffsetMode::C04_CELL_PAIRS) {
+    resultOffsetsC08.resize(ov1);
+  }
+
+  // Iteration to build the cell pairs required for the C08 base step, x, y, z reprent the spatial dimension
+  for (int x = 0; x <= overlap[0]; ++x) {
+    for (int y = 0; y <= overlap[1]; ++y) {
+      // First offset of the cell-pair, computed withhout z index
+      const int offset1 = threeToOneD(x, y, 0, cellsPerDimIntegral);
+      // The distance vector for offset1
+      const std::array<double, 3> distVec1{
+          {static_cast<double>(x) * cellLength[0], static_cast<double>(y) * cellLength[1], 0.0}};
+      for (int z = 0; z <= overlap[2]; ++z) {
+        // The z component is always calculated the in the same way
+        const double distVecZComponent = std::max(0.0, z - 1.0) * cellLength[2];
+        // Iteration over the four directions for building the cell pairs with offset1 as first partner
+        for (const C08CellDirection &direction : ALL_DIRECTIONS) {
+          // Depending on the overlap and the direction from the base cell, we skip certain pairs
+          // 1. Exclusion Criteria to skip computations by not including certain cell-combinations
+          if (includeC08Case(direction, overlap, x, y, z)) {
+            // Calculate the offset2 of the second partaking cell for interactions
+            const int offset2 = cellOffsets[calculateCell2Index(ov1, direction, z)];
+
+            // Get the multipliers, these depend on the direction and are used to calculate the distance vector for
+            // the cell with offset2
+            const auto &[xMul, yMul] = toDistVectorMultiplier(direction);
+            const std::array<double, 3> distVec2{
+                {std::max(0.0, (overlap[0] - x) * xMul + x * (1 - xMul) - 1.0) * cellLength[0],
+                 std::max(0.0, (overlap[1] - y) * yMul + y * (1 - yMul) - 1.0) * cellLength[1], distVecZComponent}};
+
+            const auto distSquare = utils::ArrayMath::dot(distVec2, distVec2);
+            // 2. Exclusion Criteria to skip computations by not including certain cell-combinations
+            //      if the distance between cell centers is actually in interactionLength
+            if (distSquare <= interactionLengthSquare) {
+              // Calculate the sorting direction if sorting is enabled
+              if constexpr (Mode == C08OffsetMode::C08_CELL_PAIRS_SORTING) {
+                const auto sortDir = computeSortingDirection(cellsPerDimIntegral, distVec1, offset2);
+                // In case of a return value, add the cell paris + their sorting direction
```

Comment:
what does "in case of a return value" mean here?

Reply:
Referenced old function behavior - I forgot to update the comment here